src/lavinmq/replication/server.cr

```diff
@@ -209,21 +223,40 @@ module LavinMQ
 
         def read_acks(socket = @socket) : Nil
           spawn action_loop, name: "Follower#action_loop"
+          @server.wait_for_followers.try_send nil
           loop do
             len = socket.read_bytes(Int64, IO::ByteFormat::LittleEndian)
             @acked_bytes += len
-            @ack.try_send nil
+            if max_lag = Config.instance.max_lag
+              if lag < max_lag
+                @ack.try_send lag
+              end
+            end
           end
         rescue IO::Error
         end
 
+        def wait_for_max_lag
+          if max_lag = Config.instance.max_lag
+            current_lag = lag
+            until current_lag < max_lag
+              select
+              when timeout(1.seconds)
+              when current_lag = @ack.receive
+              end
+              break if @server.followers.includes?(self) == false
```

Comment:
Maybe we can get rid of the timeout here by closing `@ack` in `read_acks` when the socket i closed?

Something like
```suggestion
              break unless current_lag = @ack.receive
```